src/school/service/SchoolService.ts

```diff
@@ -24,4 +24,17 @@ export class SchoolService {
     }
     this.schoolRepository.delete(schoolId);
   }
+
+  async updateSchool(id: number, cityId: number, name: string): Promise<void> {
+    const school = await this.schoolRepository.findById(id);
+    if (school === undefined) {
+      throw new Error('School not found');
+    }
+    const city = await this.schoolRepository.getByCity(cityId);
```

Comment:
En esta linea, la idea es validar si la ciudad existe o no. Para esto te toca usar el CityRepository.

```suggestion
    const city = await this.cityRepository.findById(cityId);
```